src/test_l1_dsHdmiIn.c

```diff
@@ -2008,22 +2018,20 @@ void test_l1_dsHdmiIn_negative_dsGetAllmStatus(void) {
 void test_l1_dsHdmiIn_positive_dsGetSupportedGameFeaturesList(void) {
     gTestID = 43;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
     // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
     UT_ASSERT_EQUAL(dsHdmiInInit(), dsERR_NONE);
 
     // Step 2: Call dsGetSupportedGameFeaturesList() with valid inputs (dsSupportedGameFeatureList_t*)
-    dsSupportedGameFeatureList_t supported_features_1, supported_features_2;
+    dsSupportedGameFeatureList_t supported_features_1;
     UT_ASSERT_EQUAL(dsGetSupportedGameFeaturesList(&supported_features_1), dsERR_NONE);
 
-    // Step 3: Call dsGetSupportedGameFeaturesList() with valid inputs (dsSupportedGameFeatureList_t*)
-    UT_ASSERT_EQUAL(dsGetSupportedGameFeaturesList(&supported_features_2), dsERR_NONE);
+    //step 3:Validating the results retrieved from dsGetSupportedGameFeaturesList() 
+    //and Comparing the results to the expected values defined in the test profile
+    UT_ASSERT_KVP_EQUAL_PROFILE_STRING(supported_features_1.gameFeatureList, "dsHdmiIn/gameFeatures/feature");
+    UT_ASSERT_KVP_EQUAL_PROFILE_UINT32(supported_features_1.gameFeatureCount, "dsHdmiIn/gameFeatures/count");
 
-    // Step 4: Compare the returns of steps 2/3 and make sure they compare
-    // Note: You should implement a function to compare dsSupportedGameFeatureList_t objects
-    UT_ASSERT_EQUAL(supported_features_1.gameFeatureList, supported_features_2.gameFeatureList);
-    UT_ASSERT_EQUAL(supported_features_1.gameFeatureCount, supported_features_2.gameFeatureCount);
-
-    // Step 5: Call dsHdmiInTerm() to ensure deinitialization
+    // Step 4: Call dsHdmiInTerm() to ensure deinitialization
```

Comment:
test_l1_dsHdmiIn_register -

Sync the code with the develop branch.

In the registration function, separate the source and sink test cases based on the "dsHdmiIn/Type". Additionally, compare the "dsHdmiIn/numberOfPorts"; if it is "0", skip the registration.

Reply:
Done